src/cryptonote_core/blockchain.cpp

```diff
@@ -3740,20 +3768,109 @@ bool Blockchain::check_blockchain_pruning()
   return m_db->check_pruning();
 }
 //------------------------------------------------------------------
-bool Blockchain::update_next_cumulative_weight_limit()
+void Blockchain::pop_from_long_term_block_weights()
 {
-  uint64_t full_reward_zone = get_min_block_weight(get_current_hard_fork_version());
+  m_long_term_block_weights.pop_back();
+  --m_long_term_block_weights_height;
+  if (m_long_term_block_weights_height + 1 > m_long_term_block_weights.capacity())
+  {
+    uint64_t block_height = m_long_term_block_weights_height - m_long_term_block_weights.capacity() + 1;
+    m_long_term_block_weights.push_front(block_height);
```

Comment:
Aren't we supposed to add the long term block weight corresponding to `block_height` here? Seems strange to add height to `m_long_term_block_weights`.

Reply:
Wow, it's amazing this still passes the tests :S